tests/foreman/ui_airgun/test_puppetclass.py

```diff
@@ -15,3 +15,68 @@
 
 :Upstream: No
 """
+from fauxfactory import gen_string
+from nailgun import entities
+from pytest import raises
+
+from robottelo.decorators import fixture, tier2, upgrade
+
+
+@fixture(scope='module')
+def module_org():
+    return entities.Organization().create()
+
+
+@fixture(scope='module')
+def module_loc(module_org):
+    return entities.Location(organization=[module_org]).create()
+
+
+@tier2
+@upgrade
+def test_positive_end_to_end(session, module_org, module_loc):
+    """Perform end to end testing for puppet class component
+
+    :id: f837eec0-101c-4aff-a270-652005bdee51
+
+    :expectedresults: All expected CRUD actions finished successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+    """
+    variable_name = gen_string('alpha')
+    name = gen_string('alpha')
+    hostgroup = entities.HostGroup(
+        organization=[module_org], location=[module_loc]).create()
+    puppet_class = entities.PuppetClass(name=name).create()
```

Comment:
Any specific reason for creating puppet_class using nailgun instead of creating from airgun or creating from UI front. I'm not sure how it is done from UI, probably it will be imported from if those are placed in the file system

Reply:
It is impossible to create puppet class using UI, so that is pretty much a reason =)

Reply:
that's fine !